### PR TITLE
Update docs regarding the rebase

### PR DESCRIPTION
**Pull request by kargakis**
@deads2k @liggitt @pweil- @soltysh

### CODE REVIEW

**Comment by pweil-** [comment]
:+1:

File: HACKING.md [hunk 1]
```diff
@@ -280,6 +281,8 @@ codebase to make sure the compilation is not broken, all tests pass and it's com
 refactorings, architectural changes or behavior changes introduced in Kubernetes. Make sure:
 
 1. `make clean ; hack/build-go.sh` compiles without errors and the standalone server starts correctly.
+1. all of our generated code is up to date by running all `hack/update-*` scripts.
+1. `hack/verify-open-ports.sh` runs without errors.
```
Comment (deads2k) [inline HACKING.md]: Probably need something about the upstream unit test script that adds the required files.

File: HACKING.md [hunk 2]
```diff
@@ -302,9 +305,11 @@ Place all your changes in a commit called "Refactor to match changes upstream".
 
 A typical pull request for your Kubernetes rebase will contain:
 
-1. One commit for the Godeps bump (`bump(k8s.io/kubernetes):<commit SHA>`).
-2. Zero, one or more cherry-picked commits tagged UPSTREAM.
-3. One commit "Refactor to match changes upstream".
+1. One commit for the Kuberentes Godeps bump (`bump(k8s.io/kubernetes):<commit SHA>`).
```
Comment (deads2k) [inline HACKING.md]: We agreed that it was a good idea to squash in the unit test dependencies into the bump commit right?  That keeps the history smaller.  @liggitt's sugggestion.

Comment (deads2k) [inline HACKING.md]: Should note that purely transitive dependencies get squashed into the kube bump to keep noise down.

File: HACKING.md (hunk 1 above)
Comment (kargakis) [inline HACKING.md]: added a ref about it below

File: HACKING.md (hunk 2 above)
Comment (kargakis) [inline HACKING.md]: updated to mention about squashing transitive

**Comment by kargakis** [comment]
@deads2k any other comments?

**Comment by deads2k** [comment]
> @deads2k any other comments?

No. [merge]

**Pull request merged**